crates/red_knot_server/src/server/api/notifications/did_change.rs

```diff
@@ -34,11 +34,19 @@ impl SyncNotificationHandler for DidChangeTextDocumentHandler {
             .update_text_document(&key, params.content_changes, params.text_document.version)
             .with_failure_code(ErrorCode::InternalError)?;
 
-        let db = match session.workspace_db_for_path_mut(path.as_std_path()) {
-            Some(db) => db,
-            None => session.default_workspace_db_mut(),
-        };
-        db.apply_changes(vec![ChangeEvent::file_content_changed(path)], None);
+        match path {
+            AnySystemPath::System(path) => {
+                let db = match session.workspace_db_for_path_mut(path.as_std_path()) {
+                    Some(db) => db,
+                    None => session.default_workspace_db_mut(),
+                };
+                db.apply_changes(vec![ChangeEvent::file_content_changed(path)], None);
+            }
+            AnySystemPath::SystemVirtual(virtual_path) => {
+                let db = session.default_workspace_db_mut();
+                db.apply_changes(vec![ChangeEvent::ChangedVirtual(virtual_path)], None);
```

Comment:
This is more a note to myself. It's unfortunate that we need to go through `apply_changes` here which requires going from `file -> path` but this seems necessary when they e.g. open a `pyproject.toml` (and we support linting it in the future)